packages/forklift-console-plugin/src/mock-console-extension/dynamic-plugin.ts

```diff
@@ -0,0 +1,26 @@
+import { EncodedExtension } from '@openshift/dynamic-plugin-sdk';
+import { ContextProvider } from '@openshift-console/dynamic-plugin-sdk';
+import type { ConsolePluginMetadata } from '@openshift-console/dynamic-plugin-sdk-webpack/lib/schema/plugin-package';
+
+/**
+ * The plugin will be configured only if the DATA_SOURCE is 'mock'.
+ */
+const isDataSourceMock = process.env.DATA_SOURCE === 'mock';
+
+export const exposedModules: ConsolePluginMetadata['exposedModules'] = isDataSourceMock
+  ? {
```

Comment:
still don't like this trinary thing here, we will find nicer way in a followup

Reply:
This is the way the mock extension is only included on a mock build!  If the build is not mock (be it a dev or prod build), an empty array or object is returned and has no net effect to the plugin.

Easy to verify on the output `plugin-manifest.json`.  Also you can check the generated chunks - the mock stuff just won't be included on non mock builds.  Querying the .js.map files is a good way to check:

```sh
> jq '.|{file,sources}' packages/forklift-console-plugin/dist/chunks/*.js.map
```

That will show all of the files packaged in all of the chunks (yay that the source map files are json).